jepostule/pipeline/memo.py

```diff
@@ -0,0 +1,114 @@
+import datetime
+import hmac
+import logging
+import requests
+import urllib.parse
+from requests.exceptions import ConnectionError, ReadTimeout
+from . import models
+
+
+MEMO_API_USER = 'jepostule'
+MEMO_API_KEY = 'FIXME'
+MEMO_API_URL = 'http://localhost/rest/api'
+MEMO_API_TIMEOUT = 10
+MEMO_API_JOB_APPLICATION_ENDPOINT_URL = '%s/candidature' % MEMO_API_URL
+
+
+logger = logging.getLogger(__name__)
+
+
+class TooManyRequests(Exception):
+    pass
+
+
+class RequestFailed(Exception):
+    pass
+
+
+def forward_job_application(job_application_id):
+    job_application = models.JobApplication.objects.get(id=job_application_id)
+
+    headers = {
+        'Content-Type': 'application/json',
+    }
+
+    data = {
+        "nomUtilisateur": job_application.candidate_last_name,
+        "prenomUtilisateur": job_application.candidate_first_name,
+        "emailUtilisateur": job_application.candidate_email,
+        "telUtilisateur": job_application.candidate_phone,
+        "adresseUtilisateur": job_application.candidate_address,
+        "peId": job_application.candidate_peid,
+        "romeRecherche": job_application.candidate_rome_code,
+        "lettreMotivation": job_application.message,
+        "numSiret": job_application.siret,
+    }
+
+    timestamp = make_timestamp()
+    signature = make_signature(data, timestamp, user=MEMO_API_USER)
+
+    url = '%s?user=%s&timestamp=%s&signature=%s' % (
```

Comment:
Il ne faut encoder les paramètres comme ça, mais plutôt les passer à l'argument `params` de requests https://2.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls

Reply:
Tu peux peut-être essayer de passer un `OrderedDict` à l'argument `params` :

```python
>>> from collections import OrderedDict
>>> from urllib.parse import urlencode
>>> params = OrderedDict([('user', 'toto'), ('timestamp', 1559140895), ('signature', '12313543')])
>>> urlencode(params)
'user=toto&timestamp=1559140895&signature=12313543'
```

Reply:
will do!